uber/config.py

```diff
@@ -72,6 +72,72 @@ def PREREG_BADGE_TYPES(self):
         return types
 
     @property
+    def BADGE_DISPLAY_TYPES(self):
+        """
+        There are several contexts where we want to display different badge types to select:
+        - A new attendee registering for the event
+        - An attendee claiming a blank badge in an existing group
+        - An existing attendee editing their registration
+
+        This property uses the parameters in the URL to figure out the context and return the correct
+        badge types to display for an attendee.
+        """
+        with sa.Session() as session:
+            params = cherrypy.lib.httputil.parse_query_string(cherrypy.request.query_string)
```

Comment:
Since this is effectively a method which responds to parameters, it would probably be more idiomatic to make this a method instead of a property and explicitly pass those as function parameters.  That would also make it possible to call this method from somewhere other than a page handler.